content/en/docs/tutorials/configuration/pod-sidecar-containers.md

```diff
@@ -0,0 +1,148 @@
+2. It is guaranteed that built-in sidecar containers finish AFTER all the regular containers completion.
+3. With Jobs, when Pod's `restartPolicy=OnFailure` or `restartPolicy=Never`,
+   built-in sidecar containers will not block Pod completion.
+4. Also, with Jobs, build-in sidecar containers will keep being restarted, even if regular containers would not with Pod's `restartPolicy=Never`.
+
+These benefits are drastic, and lead to faster adoption that we see with other Kubernetes features.
+
+## Adopting sidecar containers
+
+The `SidecarContainers` feature is in beta state starting Kubernetes version `1.29` and is enabled by default.
+Some clusters may have this feature disabled or have software installed that is incompatible with the feature.
+
+When this happens, the Pod may be rejected or the sidecar containers may block Pod startup, rendering the Pod useless.
+This condition is easy to detect as Pod simply get stuck on initialization. However, it is rarely clear what caused the problem.
+
+Here are the considerations and troubleshooting steps that one can take while adopting sidecar containers for their workload.
+
+**Ensure the feature gate is enabled.**
+
+As a very first step, make sure that both - API server and Nodes were upgraded to the version `1.29+`.
+The feature will break on clusters where Nodes are running earlier versions where it is not enabled.
+
+The feature gate enablement should also be validated for both - API server AND Nodes.
+
+One of the ways to check the feature gate enablement is running a command like this:
+
+- For API Server 
+  `kubectl get --raw /metrics | grep kubernetes_feature_enabled | grep SidecarContainers`
+- For the individual node: 
+  `kubectl get --raw /api/v1/nodes/<node-name>/proxy/metrics | grep kubernetes_feature_enabled | grep SidecarContainers`
+
+If you see something like: `kubernetes_feature_enabled{name="SidecarContainers",stage="BETA"} 1`,
+it means that the feature is enabled.
+
+**Check for 3rd party tooling and mutating webhooks**
+
+If you experience issues when validating the feature, it may be an indication that one of the
+3rd party tools or mutating webhooks are broken.
+
+The `SidecarContainers` feature adds a new field that is define in Kubernetes `1.28` API.
+Some tools or mutating webhooks might have been built with an earlier version of Kubernetes API.
+If tools pass the unknown fields as-is or mutating webhooks are using the recommended `Patch` method to mutate Pods, it will not be a problem.
+
+However there are tools that will strip out unknown fields and must be recompiled with the v1.28+ version of Kubernetes API.
+
+The way to check this is to use the `kubectl describe pod` command with your Pod.
+If any tools stripped out the new field (`restartPolicy:Always`), you will not find see it in the command output.
+
+If you hit an issue like this, please advise the author of the tools or the webhooks to switch to `Patch` method of modifying objects.
```

Comment:
I don't understand this advice. Webhooks do not use HTTP `PATCH`. Do you mean `"patchType": "JSONPatch"`?

The thing is, [`patchType`](https://kubernetes.io/docs/reference/config-api/apiserver-admission.v1/#admission-k8s-io-v1-PatchType) can only ever be JSON Patch; no other type is supported.

If we are writing advice for authors of API clients that do issue HTTP `PATCH` requests, are we sure that Server Side Apply or JSON Merge Patch would be unsuitable?

Please clarify the point you want to make in a review comment, and also reword accordingly?

Reply:
We can provide additional advice that is specific to people coding in Go; however, the initial / baseline advice should be agnostic about the implementation language for the webhook. We don't enforce or recommend an implementation language for admission-phase extensions.

Reply:
this is very tricky to give a proper recommendation. We see that `jsonpatch.CreatePatch` sometimes decides to generate a patch that instead of injecting a full new container, will re-inject what was the sidecar container (without the restartPolicy) and then change some fields on what was the sidecar container instead.

Container:
```
apiVersion: v1
kind: Pod
metadata:
  name: asm-sidecar-test
  namespace: default
spec:
  initContainers:
  - name: busybox-native-sidecar
    image: busybox
    restartPolicy: Always
    command:
      - "/bin/sh"
      - "-c"
      - |
        sleep infinite
  containers:
  - name: busybox
    image: busybox
    command:
      - "/bin/sh"
      - "-c"
      - |
        sleep infinite
```

Patch:

```
  {
    "op": "add",
    "path": "/spec/initContainers/1",
    "value": {
      "name": "busybox-native-sidecar",
      "image": "busybox",
      "command": [
        "/bin/sh",
        "-c",
        "sleep infinite\n"
      ],
      "resources": {}
    }
  },
  {
    "op": "replace",
    "path": "/spec/initContainers/0/name",
    "value": "istio-validation"
  },
  {
    "op": "replace",
    "path": "/spec/initContainers/0/image",
    "value": "gcr.io/istio-testing/proxyv2:latest"
  },
```

So the recommendation is basically manually craft the fields mutation

Reply:
I made the very generic statement in the doc, I am going out of my depth here. Left a comment in PR description that this is a great follow up to have. Perhaps even a separate document as it come up often in KEP's PRRs - how to handle new fields being removed.